crates/gossip/src/timeout.rs

```diff
@@ -86,9 +86,12 @@ async fn remove_timed_out_rounds(
     let mut remove = HashSet::new();
     {
         for (url, state) in accepted_round_states.iter() {
-            let state_lock = state.lock().await;
+            let Ok(state_lock) = state.try_lock() else {
```

Comment:
This is what I mean about not waiting on both locks. If we can check this session then do. If it's currently in use then clearing it out will have to wait.